File: api/ranking.py
import os
import json
from pymongo import MongoClient
from bson import json_util

headers = {
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Methods': 'GET, POST, OPTIONS',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Content-Type': 'application/json'
}

def handler(request):
    if request.method == "OPTIONS":
        return "", 200, headers

    try:
        mongo_uri = os.environ.get("MONGODB_URI")

        if not mongo_uri:
            return json.dumps({"erro": "Variável MONGO_URI não definida"}), 500, headers

        client = MongoClient(mongo_uri)
        db = client['seu_banco_de_dados']
        ranking_collection = db['ranking']

        if request.method == "POST":
            data = json.loads(request.body)
            nome = data.get('nome')
            pontuacao = int(data.get('pontuacao'))

            if nome and pontuacao is not None:
                ranking_collection.insert_one({'nome': nome, 'pontuacao': pontuacao})
                return json.dumps({"mensagem": "Pontuação salva com sucesso!"}), 200, headers
            return json.dumps({"erro": "Dados inválidos"}), 400, headers

        elif request.method == "GET":
            top_5_results = ranking_collection.find().sort("pontuacao", -1).limit(5)
            ranking_json = json.loads(json_util.dumps(top_5_results))
            return json.dumps(ranking_json), 200, headers

        else:
            return json.dumps({"erro": "Método não permitido"}), 405, headers

    except Exception as e:
        import traceback
        traceback.print_exc()
        return json.dumps({"erro": f"Erro interno: {e}"}), 500, headers

File: api/ranking.js
const { MongoClient } = require('mongodb');

module.exports = async (request, response) => {
  response.setHeader('Access-Control-Allow-Origin', '*');
  response.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  response.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (request.method === 'OPTIONS') {
    return response.status(200).end();
  }

  try {
    const mongo_uri = process.env.MONGO_URI;

    if (!mongo_uri) {
      return response.status(500).json({ error: 'Variável MONGO_URI não definida' });
    }

    const client = new MongoClient(mongo_uri);
    await client.connect();

    const db = client.db('pontuacaodados');
    const rankingCollection = db.collection('ranking');

    if (request.method === 'POST') {
      const { nome, pontuacao } = request.body;
      if (nome && pontuacao !== undefined) {
        await rankingCollection.insertOne({ nome, pontuacao });
        return response.status(200).json({ mensagem: 'Pontuação salva com sucesso!' });
      }
      return response.status(400).json({ erro: 'Dados inválidos' });
    }

    if (request.method === 'GET') {
      const top5 = await rankingCollection.find().sort({ pontuacao: -1 }).limit(5).toArray();
      return response.status(200).json(top5);
    }

    return response.status(405).json({ erro: 'Método não permitido' });
  } catch (e) {
    console.error(e);
    return response.status(500).json({ erro: `Erro interno do servidor: ${e.message}` });
  }
};